applications/GeoMechanicsApplication/tests/cpp_tests/test_apply_k0_procedure_process.cpp

```diff
@@ -89,26 +85,14 @@ namespace
 
 ModelPart& PrepareTestModelPart(Model& rModel)
 {
-    auto& result = rModel.CreateModelPart("dummy");
-
-    // Set up the test model part mesh
-    auto                   p_point_1 = Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0);
-    auto                   p_point_2 = Kratos::make_intrusive<Node>(2, 0.0, 1.0, 0.0);
-    auto                   p_point_3 = Kratos::make_intrusive<Node>(3, 1.0, 1.0, 0.0);
-    auto                   p_point_4 = Kratos::make_intrusive<Node>(4, 1.0, 0.0, 0.0);
-    Quadrilateral2D4<Node> domain_geometry(p_point_1, p_point_2, p_point_3, p_point_4);
-
-    Parameters mesher_parameters(R"({
-        "number_of_divisions": 2,
-        "element_name": "Element2D3N",
-        "condition_name": "LineCondition",
-        "create_skin_sub_model_part": true
-    })");
-    StructuredMeshGeneratorProcess(domain_geometry, result, mesher_parameters).Execute();
-
+    auto& result                  = rModel.CreateModelPart("dummy");
     auto  p_dummy_law             = std::make_shared<Testing::StubLinearElasticLaw>();
-    auto& r_model_part_properties = result.GetProperties(0);
-    r_model_part_properties.SetValue(CONSTITUTIVE_LAW, p_dummy_law);
+    auto  p_model_part_properties = result.pGetProperties(0);
+    p_model_part_properties->SetValue(CONSTITUTIVE_LAW, p_dummy_law);
+
+    auto p_element = make_intrusive<StubElement>();
```

Comment:
If you want to (it's really not necessary to complete this PR, but just if you're interested), you can try to refactor the stub element, such that it uses gmock (i.e. using the MOCK_METHOD macro). For an example of this, you could have a look at https://github.com/KratosMultiphysics/Kratos/pull/12841/files. 

For example in this test: https://github.com/KratosMultiphysics/Kratos/pull/12841/files#diff-e2152fdb220b33efa4faa1c5ee19176a719a6500463157955e737fc941697f64R71-R81 you can see how you can use a mocked method to return a certain value.

Again, it's not needed for this PR, it's just if you're interested in GMock, you can have a go :)

Reply:
Thank you for the suggestion. I think there is no need to change the current `StubElement `but I used mock for unit tests for `K0_MAIN_DIRECTION`.